jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py

```diff
@@ -88,13 +91,25 @@ def __init__(
         dump_path = dump_path or kwargs.get('runtime_args').get('dump_path')
         if dump_path is not None:
             self.logger.info('Start building "FaissIndexer" from dump data')
-            ids, vecs = import_vectors(dump_path, str(self.runtime_args.pea_id))
-            self._ids = np.array(list(ids))
+            ids_iter, vecs_iter = import_vectors(
+                dump_path, str(self.runtime_args.pea_id)
+            )
+            self._ids = np.array(list(ids_iter))
             self._ext2int = {v: i for i, v in enumerate(self._ids)}
-            self._vecs = np.array(list(vecs))
-            self.num_dim = self._vecs.shape[1]
-            self.dtype = self._vecs.dtype
-            self.index = self._build_index(self._vecs)
+
+            self._prefech_data = []
+            if self.prefech_size and self.prefech_size > 0:
```

Comment:
So if you have a dump of size N and prefetch of size X (X < N), how do you build the entire N index? It seems here that you're limiting to just X

Reply:
ah no, nvm, got it

Reply:
This is only the first prefetch step. The full fetch procedure happens in `self._build_partial_index` function